### ISSUE TITLE
6.0.0rc4 breaks pygmt

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#1636]
Below is a simple PyGMT script, which used to work with 6.0.0rc1.
```
import pygmt
fig = pygmt.Figure()
fig.basemap(region="0/10/0/10", frame=True, projection="X10c")
fig.savefig("map.pdf")
```

The PyGMT script is equivalent to the following bash script:
```
gmt begin pygmt-session
gmt figure unique_name -
gmt basemap -R0/10/0/10 -Baf -JX10c
gmt psconvert -A -Fmap -Qg2 -Qt2 -Tf
gmt end
```
Running the bash script above gives following error message:
```
psconvert [ERROR]: Syntax error: Modern GMT mode requires the -F option
end [ERROR]: Failed to call psconvert
end [ERROR]: process_figures returned error 71
```

According to the [comments in PyGMT codes](https://github.com/GenericMappingTools/pygmt/blob/34f5291a7d756ab3edd97be53e6633f3ee9596f1/pygmt/figure.py#L86), passing format `-` to the **figure** module means "don't produce files for this **figure** when running **gmt end**". It seems it doesn't work correctly in 6.0.0rc4.

**Comment by weiji14** [comment]
The '-' [format](https://docs.generic-mapping-tools.org/latest/begin.html#tbl-formats) doesn't seem to be documented in the [`figure`](https://github.com/GenericMappingTools/gmt/blob/146e5b9378caa5b1bb8e87de9721444112341177/doc/rst/source/figure.rst#optional-arguments) docs either.

**Comment by PaulWessel** [comment]
Are you sure that is how Leo used it? The C code checks to see if the _figure name_ equals -, not the format.  And since that fails it tries something...  I can add a check so that if either name or format equals - then skip it.

**Comment by seisman** [comment]
Closed as fixed by #1638.
